map.go

```diff
@@ -76,6 +76,18 @@ func PickByValues[K comparable, V comparable, Map ~map[K]V](in Map, values []V)
 	return r
 }
 
+// ValuesByKeys returns an array of values in the same order as the given keys.
+// Play: https://go.dev/play/p/PmmPDo1AqWl
+func ValuesByKeys[K comparable, V any](in map[K]V, keys []K) []V {
+	out := make([]V, 0, len(keys))
+	for i := range keys {
+		if v, ok := in[keys[i]]; ok {
```

Comment:
Just a random doubt: If the key is not present in the map, then the order in the output slice will not be aligned to the input keys.
You can add that edge case in the example of Readme. 
Your thoughts on it ??

Reply:
Thanks for your suggestion!
I agree with you to build this edge case into the example. 
Another way would be to append nil or zero values to the slice, but this solution requires additional checking by the caller, which I don't think is a good practice.
WDYT?

Reply:
I agree with @shivamrazorpay 

It's the first thing I thought about

I would expect to return an error in such case

Reply:
That seems reasonable.

I have resubmitted the code